packages/dds/sequence/src/test/reentrancy.spec.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { MergeTreeDeltaType } from "@fluidframework/merge-tree";
+import {
+	MockFluidDataStoreRuntime,
+	MockContainerRuntimeFactory,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { SharedString } from "../sharedString";
+
+describe("SharedString op-reentrancy", () => {
+	let sharedString: SharedString;
+	let sharedString2: SharedString;
+	let containerRuntimeFactory: MockContainerRuntimeFactory;
+	beforeEach(() => {
+		containerRuntimeFactory = new MockContainerRuntimeFactory();
+
+		const factory = SharedString.getFactory();
+		const dataStoreRuntime1 = new MockFluidDataStoreRuntime();
+		dataStoreRuntime1.local = false;
+
+		sharedString = factory.create(dataStoreRuntime1, "A");
+
+		const containerRuntime1 = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime1);
+		const services1 = {
+			deltaConnection: containerRuntime1.createDeltaConnection(),
+			objectStorage: new MockStorage(),
+		};
+		sharedString.connect(services1);
+
+		const dataStoreRuntime2 = new MockFluidDataStoreRuntime();
+		dataStoreRuntime2.local = false;
+		const containerRuntime2 = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime2);
+		const services2 = {
+			deltaConnection: containerRuntime2.createDeltaConnection(),
+			objectStorage: new MockStorage(),
+		};
+
+		sharedString2 = factory.create(dataStoreRuntime2, "B");
+		sharedString2.connect(services2);
```

Comment:
I don't actually use the second sharedString here. I could remove it now, but pretty sure we'll want to re-add it later for upcoming cases where we do support reentrancy (op processing causes an op to be submitted).